tensorflow/compiler/xla/service/hlo_computation.cc

```diff
@@ -919,7 +937,11 @@ Status HloComputation::ReplaceInstruction(HloInstruction* old_instruction,
       ShapeUtil::Compatible(old_instruction->shape(), new_instruction->shape()))
       << ShapeUtil::HumanString(old_instruction->shape()) << " vs "
       << ShapeUtil::HumanString(new_instruction->shape());
+  return ReplaceInstructionWithDifferentShape(old_instruction, new_instruction);
+}
 
+Status HloComputation::ReplaceInstructionWithDifferentShape(
```

Comment:
I feel quite uneasy about this API, how does it work when shapes don't match? Could you give an example?

Reply:
In the file `hlo_instruction.h`, there is already many `Replace*WithDifferentShape()` like `ReplaceAllUsesWithDifferentShape()`. I'm just following the same pattern. I do this in the hlo_computation.h files as the nodes are in an HloComputation and I want its internal structure to show the changes. Doing the update manually would be brittle and would need more code of line.

Here I use it to lift a bitcast. Example before:
```
%1=tanh(f32[32,32,32] input)
%2=f32[1024,32] bitcast(%1)
```
after:
```
%3=f32[1024,32] bitcast(f32[32,32,32] input)
%4=tanh(%1)
```
I need to replace the %1 by %3 and they have different shape.
This is a powerful function that must be used with care. Like all the `HloInstruction::Replace*WithDifferentShape()` functions. This is why I limit the ops supported inside the fusion.